src/field_theory/ax_grothendieck.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2023 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+import data.mv_polynomial.basic
+import ring_theory.algebraic
+import data.fintype.card
+import data.zmod.basic
+
+/-!
+# Ax-Grothendieck for algebraic extensions of `zmod p`
+
+This file proves that if `R` is an algebraic extension of a finite field,
+then any injective polynomial map `R^n -> R^n` is also surjective.
+
+This proof is required for the true Ax-Grothendieck theorem, which proves the same result
+for any algebraically closed field of characteristic zero. The proof of the theorem for
+characteristic zero is not in mathlib, but it is at https://github.com/Jlh18/ModelTheoryInLean8
+-/
+
+noncomputable theory
+
+/-- A polynomial map over an indexing type `ι` is a vector of `ι` polynomials
+each in `ι` variables -/
+@[simp]
+def poly_map (K : Type*) [comm_semiring K] (ι  : Type*) : Type* :=
+ι → mv_polynomial ι K
```

Comment:
What's the motivation for defining this type synonym? to have a custom `mv_polynomial.eval` in a nice form? If so, I think you could switch to notation for `ι → mv_polynomial ι K` + notation for `λ ps as k, mv_polynomial.eval as (ps k)`. That would also avoid dumping an ambiguous name in `_root_` (I can think of at least three different mathematical objects that one could reasonably call `poly_map`).